Translate FAQ AJAX and APIs (#35)
reactjs/id.reactjs.org: content/docs/faq-ajax.md

---
id: faq-ajax
title: AJAX dan API
permalink: docs/faq-ajax.html
layout: docs
category: FAQ
---

### Bagaimana cara membuat panggilan AJAX? {#how-can-i-make-an-ajax-call}

> Anda bisa menggunakan semua AJAX *library* yang anda inginkan seperti halnya dengan React. Salah satu diantaranya yang populer yaitu [Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/), dan *browser* yang ter-integrasi [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).

Reviewer: > Anda dapat menggunakan semua _library_ AJAX yang diinginkan pada React. Beberapa yang populer yaitu Axios, jQuery AJAX, dan `window.fetch` bawaan browser.

Reviewer 2: > Anda dapat menggunakan *library* AJAX apa saja untuk React.

Just a minor thing, *apa saja* for *any* instead of *semua*.

Reviewer 3: ```suggestion
Anda bisa menggunakan *library* AJAX apa saja yang anda inginkan dengan React. Beberapa yang populer yaitu [Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/), dan [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) bawaan *browser*.
```

> ### Dimana komponen *lifecycle* harus menggunakan panggilan AJAX? {#where-in-the-component-lifecycle-should-i-make-an-ajax-call}

Reviewer: > Pada siklus komponen bagian mana seharusnya membuat panggilan AJAX?

Reviewer 3: ```suggestion
### Pada *lifecycle* komponen mana seharusnya membuat panggilan AJAX? {#where-in-the-component-lifecycle-should-i-make-an-ajax-call}
```

> Anda seharusnya dapat mengisi data melalui panggilan AJAX di dalam [`componentDidMount`](/docs/react-component.html#mounting) *lifecycle method*. Disini anda juga bisa menggunakan `setState` untuk memperbarui komponen anda ketika data telah diterima.

Reviewer: > Anda sebaiknya mengisi data dengan panggilan AJAX pada _method_ `componentDidMount`. Hal ini agar Anda dapat menggunakan `setState` untuk memperbarui komponen saat data diambil.

Reviewer 3: ```suggestion
Anda sebaiknya mengisi data melalui panggilan AJAX pada *lifecycle method* [`componentDidMount`](/docs/react-component.html#mounting). Dengan begitu Anda dapat menggunakan `setState` untuk memperbarui komponen Anda ketika data telah diterima.
```

> ### Contoh: Penggunaan AJAX yang ditentukan ke *state* lokal {#example-using-ajax-results-to-set-local-state}

Reviewer 2: > Menggunakan hasil AJAX untuk (mengatur/menentukan) *state* lokal.

I fee like **to** is a verb here.

> ### Contoh: Penggunaan AJAX yang ditentukan ke *state* lokal {#example-using-ajax-results-to-set-local-state}

Reviewer 3: ```suggestion
### Contoh: Menggunakan hasil AJAX untuk mengisi *state* lokal {#example-using-ajax-results-to-set-local-state}
```

What do you think? 😄

> Komponen dibawah menunjukan cara membuat panggilan di AJAX  `componentDidMount` untuk mengisi komponen *state* lokal  .

Reviewer 4: > Komponen dibawah menunjukkan cara menggunakan panggilan AJAX pada `componentDidMount` untuk mengisi *state* komponen lokal.

> Contoh penggunaan API yang dikembalikan ke objek JSON seperti ini:

Reviewer 4: > contoh penggunaan API memberikan objek JSON seperti ini:

Reviewer 2: > Contoh API (berikut/dibawah) ini menghasilkan objek JSON (sebagai/seperti) berikut:

It seems to me *The example API* is a noun that refer to `fetch("https://api.example.com/items")` in the code block.

            items: result.items
          });
        },

>         // catatan: sangatlah penting untuk mengatasi kesalahan disini

Reviewer 3: ```
// Catatan: sangatlah penting untuk mengatasi error disini
// alih-alih menggunakan blok catch() sehingga kita tidak menenggelamkan
// exception dari bug yang sebenarnya terjadi di komponen
```

What do you think? 😄

>         // alih-alih menggunakan blok catch() sehingga kita tidak menenggelamkan

Reviewer: ```suggestion
        // daripada menggunakan blok catch() sehingga kita tidak menenggelamkan
```

Reviewer: @resir014 or @regalius?

Reviewer 5: Yep, let's go with your suggestion.
